### PR TITLE
Move JDBC connector deps to :dev alias

### PR DESCRIPTION
**Pull request by PavlosMelissinos**

### CODE REVIEW

**Comment by PavlosMelissinos** [comment]
konserve-jdbc didn't have a dev profile, so I made one.

FYI, the dependencies will obviously have to be added to the dev alias every library that uses konserve-jdbc

**Comment by whilo** [comment]
@TimoKramer @jsmassa Does that make sense to you?

**Comment by TimoKramer** [comment]
And some update in the documentation would be good where you describe the breaking change and the need to add the drivers manually

**Comment by PavlosMelissinos** [comment]
> We'll need the drivers in the test-alias as well otherwise the tests won't run.

In theory you can use something like `clj -M:dev:test` to stack the aliases but I'm not sure how you actually run the tests (you're using an orb, right?)

> And some update in the documentation would be good where you describe the breaking change and the need to add the drivers manually

Good point!

Would a note in README.md suffice? It would be nice to be included in the next release notes too.

Also are you sure you're ok with introducing this breaking change? I suppose most users will provide their own jdbc driver(s) as direct dependencies, so few apps will actually break but you never know...

**Comment by TimoKramer** [comment]
> In theory you can use something like `clj -M:dev:test` to stack the aliases but I'm not sure how you actually run the tests (you're using an orb, right?)

right, I am hesitant to change the orb

> Would a note in README.md suffice? It would be nice to be included in the next release notes too.

Right, we'll add it prominently to the README and we'll put it in the commit message and we'll announce it on our slack channel.

> Also are you sure you're ok with introducing this breaking change? 
It's the right way to go so I am for it but I guess some people are relying on our dependencies. Still we were not good so far with keeping the deps up to date so most users should have overwritten them anyway.

**Comment by PavlosMelissinos** [comment]
Alright @TimoKramer , please check again, I think I've addressed everything!

By the way I ran `./bin/run-integrationtests` from the main branch but it fails. Something is off with networking because it doesn't "see" the postgres container for some reason.

Apparently it's working for you so it shouldn't matter, it just means that I can't test that everything still works as expected.

File: README.md [hunk 1]
```diff
@@ -76,8 +76,12 @@ Add to your dependencies:
 
 ## Supported Databases
 
-Not all databases available for JDBC have been tested to work with this implementation. 
-Other databases might still work, but there is no guarantee. 
+**BREAKING CHANGE**: konserve-jdbc versions after `0b68d31` no longer include
```
Comment (TimoKramer) [inline README.md:79]:
```suggestion
**BREAKING CHANGE**: konserve-jdbc versions after `0.1.79` no longer include
```

File: README.md [hunk 2]
```diff
@@ -76,8 +76,12 @@ Add to your dependencies:
...
+actual JDBC drivers. Before you upgrade please make sure your application
+provides the necessary dependencies.
+
+Not all databases available for JDBC have been tested to work with this implementation.
+Other databases might still work, but there is no guarantee.
```
Comment (TimoKramer) [inline README.md:84]:
```suggestion
Other databases might still work, but there is no guarantee. Please see working
drivers in the dev-alias in the `deps.edn` file.
```

**Review by TimoKramer — APPROVED** [review]
I think we are good. Just minor changes in wording. @alekcz @jsmassa What do you think?

**Comment by TimoKramer** [comment]
One thing you @PavlosMelissinos could change is to increase the minor release version in the build.clj to 0.2.

**Comment by PavlosMelissinos** [comment]
@TimoKramer done, I think

**Comment by TimoKramer** [comment]
Merge? @whilo

**Comment by whilo** [comment]
Thanks a lot for the PR!

**Pull request merged**